File: devcamp2018/app/src/main/java/com/psato/devcamp/di/component/BaseFragmentComponent.java
package com.psato.devcamp.di.component;

import com.psato.devcamp.di.FragmentScoped;
import com.psato.devcamp.presentation.base.BaseFragment;

import dagger.Component;


@FragmentScoped
@Component(dependencies = {ApplicationComponent.class})
public interface BaseFragmentComponent {
    void inject(BaseFragment baseFragment);
}

File: devcamp2018/app/src/main/java/com/psato/devcamp/di/component/BaseFragmentComponent.kt
package com.psato.devcamp.di.component

import com.psato.devcamp.di.FragmentScoped
import com.psato.devcamp.presentation.base.BaseFragment

import dagger.Component


@FragmentScoped
@Component(dependencies = arrayOf(ApplicationComponent::class))
interface BaseFragmentComponent {
    fun inject(baseFragment: BaseFragment)
}
